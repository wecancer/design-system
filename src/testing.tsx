import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
