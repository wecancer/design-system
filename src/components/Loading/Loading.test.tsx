import Loading from '.'
import {render} from '../../testing'

describe('<Loading />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <Loading />
        <Loading size="2.5rem" />
        <Loading title="custom title" />
        <Loading type="dots" />
        <Loading type="solid" />
        <Loading color="primary" />
        <Loading color="white" />
        <Loading color="error" />
        <Loading color="success" />
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
