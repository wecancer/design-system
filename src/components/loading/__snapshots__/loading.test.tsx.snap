// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render snapshot correctly 1`] = `
<div
  class="sc-gsDKAQ cbKTir"
>
  <div
    class="sc-bdvvtL bYGNWJ"
    data-icon-name="spinnerSolid"
  >
    <div>
      mock svg
    </div>
  </div>
</div>
`;
